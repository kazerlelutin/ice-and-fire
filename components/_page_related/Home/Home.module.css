.container {
    display: flex;
    flex-direction: column;
    gap: 15px;
}

.events {
    display: flex;
    gap: 15px;
}