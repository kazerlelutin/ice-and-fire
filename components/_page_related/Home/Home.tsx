
import classes from './Home.module.css'
import CommonLayout from '../../_layouts/CommonLayout/CommonLayout'
import Books from '../../Books/Books'

export default function Home() {
  return (
    <CommonLayout>
      <div className={classes.container}>
      <Books/>
      </div>
    </CommonLayout>
  )
}
