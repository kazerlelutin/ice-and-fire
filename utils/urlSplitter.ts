//for extract id 
export default function urlSplitter(url:string):string {
  const splitUrl = url.split('/')
  return splitUrl.at(-1)
}