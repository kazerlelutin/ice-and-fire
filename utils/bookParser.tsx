import { IapiBook } from "../types/apiBook.interface";
import { IBook } from "../types/book.interface";
import urlSplitter from "./urlSplitter";

export default function bookParser(book:IapiBook):IBook{

  return {
    id:urlSplitter(book.url),
    isbn: book.isbn,
    name: book.name,
    released: book.released,
    numberOfPages: book.numberOfPages,
    //merge all type of characters, just IDs
    characters: [...book.characters, ...book.povCharacters].reduce(
      (previousValue, currentValue) => {
        const 
        characterId = urlSplitter(currentValue),
        isExist = previousValue.find((id) => id === characterId)
        if (!isExist) previousValue.push(characterId)
        return previousValue
      },
      []
    ),
  }
}