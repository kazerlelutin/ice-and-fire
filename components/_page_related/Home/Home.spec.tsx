import { afterEach, describe, expect, it } from 'vitest'
import {cleanup } from '@testing-library/react'
import Home from './Home'
import { render} from '../../../utils/test-utils'

describe('<Home />', () => {
  afterEach(cleanup)

  it('Check snapshot', async () => {
    const { asFragment } = render(<Home/>)
    expect(asFragment()).toMatchSnapshot()
  })
})


