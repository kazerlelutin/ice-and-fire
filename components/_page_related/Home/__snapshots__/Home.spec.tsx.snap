// Vitest Snapshot v1

exports[`<Home /> > Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_f3f7d0"
  >
    <header
      class="_header_f3f7d0"
    >
      <div
        class="_container_46c4bf"
      >
        <a
          href="/"
        >
          <div
            class="_container_5bb206"
          >
            Ice & Fire
          </div>
        </a>
      </div>
    </header>
    <main
      class="_main_f3f7d0"
    >
      <div
        class="_container_99e338"
      >
        <div
          class="_container_a73fd7"
        >
          <div
            class="_books_a73fd7"
          />
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
